### PR TITLE
DOCSP-41333 Version Release for 3.0 and 3.0.1

### PR DESCRIPTION
**Pull request by shuangela**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-rust/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-41333>
Staging - <https://preview-mongodbshuangela.gatsbyjs.io/rust/DOCSP-41333-rust-3.0-update/>

This pr includes release notes/compat table/updates for both 3.0 and 3.0.1.
## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/whats-new.txt [hunk 1]
```diff
@@ -20,11 +20,31 @@ What's New
...
+.. _rust-version-3.0.1:
+
+What's New in 3.0.1
+---------------------
+
+The {+driver-short+} v3.0.1 release includes the following features,
+improvements, and fixes:
+
+- Fixes issue where aggregations with both explicit sessions and custom types
+  wouldn't compile.
```
Comment (rustagir) [inline source/whats-new.txt:38]:
```suggestion
- Fixes an issue in which aggregations run within explicit sessions and aggregation
  that used custom types wouldn't compile.
```

File: source/whats-new.txt [hunk 2]
```diff
@@ -20,11 +20,31 @@ What's New
...
+
+- Fixes issue where ``Collection::watch`` didn't use the ``Collection`` type
+  parameter.
```
Comment (rustagir) [inline source/whats-new.txt:41]:
```suggestion
- Fixes an issue in which the ``Collection::watch()`` method didn't
  use the type parameter of the ``Collection`` instance it was run on.
```

File: source/whats-new.txt [hunk 3]
```diff
@@ -20,11 +20,31 @@ What's New
...
+
+To learn more about this release, see the
+`v3.0.1 Release Notes
+<https://github.com/mongodb/mongo-rust-driver/releases/tag/v3.0.1>`__
```
Comment (rustagir) [inline source/whats-new.txt:45]: S: let's move `https://github.com/mongodb/mongo-rust-driver/` into a source constant and replace the link stem throughout the page with that constant

Comment (rustagir) [reply source/whats-new.txt:45]: It can be called "driver-source"

**Review by rustagir — CHANGES_REQUESTED** [review]
A few small changes!

File: source/whats-new.txt (hunk 1 above)
Comment (shuangela) [reply source/whats-new.txt:38]: Based on the PR for this change, it seems like the issue is: "Aggregation with both an explicit session and a custom type isn't possible. While helpers for both with_type and session exist in the 3.0 API, it's impossible to use both on the same aggregation." I think the current suggestion seems to imply it's that both explicit sessions and custom types don't work individually. What do you think of:

Fixes an issue in which aggregations run within explicit sessions which also used custom types wouldn't compile.

Comment (rustagir) [reply source/whats-new.txt:38]: sounds good!

File: snooty.toml [hunk 4]
```diff
@@ -21,11 +21,12 @@ driver-long = "MongoDB Rust Driver"
...
+version = "3.0.1" # always set this to the driver version (i.e. 2.6.0, 2.5.0, etc.)
 min-rust-version = "1.60" # always set this to the minimum supported Rust version
 api = "https://docs.rs/mongodb/{+version+}/mongodb"
-bson-version = "2.8.0"
+bson-version = "2.11.0"
 bson-api = "https://docs.rs/bson/{+bson-version+}/bson"
 stable-api = "Stable API"
 tracing-version = "0.1.37"
 tracing-sub-version = "0.3.17"
+driver-source = "https://github.com/mongodb/mongo-rust-driver/"
```
Comment (rustagir) [inline snooty.toml:32]: nit: remove the trailing slash at the end and adjust the usages of the source constant. This creates cleaner looking links than having the rest of the link directly abut the source constant brackets:
`{+constant+}this/is/my/link` vs `{+constant+}/this/is/my/link`

**Review by rustagir — APPROVED** [review]
lgtm save a few small remaining suggestions!

**Pull request merged**